apiserver/facades/controller/crossmodelsecrets/crossmodelsecrets.go

```diff
@@ -263,8 +265,13 @@ func (s *CrossModelSecretsAPI) canRead(ctx stdcontext.Context, secretService Sec
 	}
 
 	appName, _ := names.UnitApplication(unit.Id())
+	k := secretservice.ApplicationAccessor
```

Comment:
can we have a better name?

Reply:
Which bit? "Accessor"? It is the entity which is wanting to access the secret. And we have Applications, Units, etc which are the concrete entities. It makes sense to me - I can't think of anything better but happy for suggestions.